src/areas/mod.rs

```diff
@@ -136,8 +136,43 @@ impl Areas {
         }
         Areas {
             areas,
-            touch_width: touch_width as u32,
-            touch_height: touch_height as u32,
+            touch_width,
+            touch_height,
+        }
+    }
+
+    pub fn parallelograms(
+        touch_width: i32,
+        touch_height: i32,
+        (width, height): (i32, i32),
+        slantiness: i32,
+        row_interval: i32,
+    ) -> Areas {
+        let mut areas = vec![];
+        let ncols = (touch_width as f32 / width as f32).ceil() as i32;
+        let nrows = (touch_height as f32 / height as f32).ceil() as i32;
+        for row in 0..nrows {
```

Comment:
I feel like the variable naming is a bit inconsistent with the rest of the codebase, where I tried to avoid any abbreviations. So I would suggest to rename `col` to `columns` and `ncols` to `number_of_columns` and `nrows` to `number_of_rows`. @caroline-lin: Any opinions?

Reply:
Ok! I don't have a strong opinion on this.